Shared (App)/Services/TransactionInputParser.swift

```diff
@@ -22,21 +25,51 @@ final class TransactionInputParser: TransactionInputParseable {
               let data = try? Data(contentsOf: url),
               let methodInfo = try? decoder.decode([MethodInfo].self, from: data) else { return [:] }
         
-        let methodInfoDict = methodInfo.reduce([RawAddress: [MethodID: MethodInfo]]()) { partialResult, info in
-            var methodInfoDict = partialResult
-            if methodInfoDict[info.contractAddress] != nil {
-                methodInfoDict[info.contractAddress]?[info.methodId] = info
+        return methodInfo.reduce([RawAddress: [MethodID: MethodInfo]]()) { partialResult, info in
+            var methodInfo = partialResult
+            if methodInfo[info.contractAddress] != nil {
+                methodInfo[info.contractAddress]?[info.methodId] = info
             } else {
-                methodInfoDict[info.contractAddress] = [info.methodId: info]
+                methodInfo[info.contractAddress] = [info.methodId: info]
             }
-            return methodInfoDict
+            return methodInfo
         }
-        return methodInfoDict
     }()
     
     func parse(input: [String: String], methodHash: String, contractAddress: RawAddress) -> String? {
         guard let methodInfo = methodInfoLookup[contractAddress]?[methodHash] else { return nil }
         let arguments = methodInfo.description.arguments.compactMap { input[$0] }
         return String(format: methodInfo.description.stringFormat, arguments: arguments)
     }
+    
+    func parseToStringValues(input: TransactionInput) -> [InputName: String] {
+        input.inputData.reduce([InputName: String]()) { partialResult, input in
+            var stringValues = partialResult
+            stringValues[input.key] = convertRawValueToString(inputData: input.value)
+            return stringValues
+        }
+    }
+    
+    private func convertRawValueToString(inputData: InputData) -> String? {
```

Comment:
Just simple conversion to string values right now so we can display it on the tx history detail page for demo purposes.